### PR TITLE
Feature/gh2 enhance backends

### PR DESCRIPTION
**Pull request by bheisen**
This PR gives the possibility to deal also with an entire set of (named) instances as provided by an VRPC agent. The VrpcBackendMaker will track the life-time and potential event emissions of existent backend instances.

### CODE REVIEW

File: src/react-vrpc.js [hunk 1]
```diff
@@ -28,26 +31,123 @@ class VrpcBackendMaker extends Component {
...
+          default:
+            obj[key].push({ ...proxy, [event]: args })
+        }
+        this.setState({ [key]: obj[key] })
+      })
+    }
+  }
+
   render () {
+    const { loading } = this.props
```
Comment (benjamin-joham) [inline src/react-vrpc.js:148]: Where does this loading prop come from?

**Review by benjamin-joham — COMMENTED** [review]
Besides the loading prop in [src/react-vrpc.js#L148](https://github.com/JacobTheEvans/react-vrpc/blob/feature/GH2-enhance-backends/src/react-vrpc.js#L148) everything looks clear and works nicely.

File: src/react-vrpc.js [hunk 2]
```diff
@@ -4,17 +4,20 @@ import { VrpcRemote } from 'vrpc'
 const VrpcContext = React.createContext()
```
Comment (JacobTheEvans) [inline src/react-vrpc.js:4]: Noting in this file our discussion outside of github. We will merge in this PR but have a new ticket made where we tie the lifecycle of the backend agents, classes and instances all the component lifecycle methods. We are think of using something similar to the react-apollo project

**Review by JacobTheEvans — APPROVED** [review]
We will merge in this PR but have a new ticket made where we tie the lifecycle of the backend agents, classes and instances all the component lifecycle methods. We are think of using something similar to the react-apollo project

**Pull request merged**